types/json-schema/index.d.ts

```diff
@@ -12,8 +12,7 @@
 /**
  * @see https://tools.ietf.org/html/draft-zyp-json-schema-03#section-5.1
  */
-export type JSONSchema4TypeName = 'string' | 'number' | 'integer' | 'boolean'
-    | 'object' | 'array' | 'null' | 'any'
+export type JSONSchema4TypeName = string | number | boolean | object | null | any
```

Comment:
As mentioned by @cyrilletuzi the type is indeed an enum of string values, not their corresponding typescript types